modules/BL/businessesBL.js

```diff
@@ -50,15 +50,71 @@ let self = module.exports = {
 
     GetWorkersForBusiness(businessId) {
         return new Promise((resolve, reject) => {
-            DAL.FindOne(businessesCollectionName, { _id: DAL.GetObjectId(businessId) })
-                .then(business => {
-                    let workers = business.workers;
-                    workers.push(business.manager);
+            let businessFilter = { $match: { "_id": DAL.GetObjectId(businessId) } };
+            let joinWorkersQuery = {
+                $lookup:
+                {
+                    from: usersCollectionName,
+                    localField: 'workers',
+                    foreignField: '_id',
+                    as: 'workers'
+                }
+            }
 
-                    DAL.FindSpecific(usersCollectionName, { _id: { $in: workers } })
-                        .then(resolve)
-                        .catch(reject);
-                }).catch(reject);
+            let joinManagerQuery = {
+                $lookup:
+                {
+                    from: usersCollectionName,
+                    localField: 'manager',
+                    foreignField: '_id',
+                    as: 'manager'
+                }
+            }
+
+            let projectFilter = {
+                $project: {
+                    "workers": 1,
+                    "manager": 1
+                }
+            }
+
+            let aggregatePipline = [
+                businessFilter,
+                joinWorkersQuery,
+                joinManagerQuery,
+                projectFilter
+            ]
+
+            DAL.Aggregate(businessesCollectionName, aggregatePipline).then(result => {
+                let business = result[0];
+                let allWorkers = business.workers.concat(business.manager);
```

Comment:
למה לא push?

Reply:
כי join aggregate מחזיר מערך וחיבור של שני מערכים עושים עם concat

Reply:
Concat זה כדי לשרשר שני מערכים
workers, manager הם מערכים?
manager זה פשוט ה id שלו

Reply:
join aggregate returns array even for single value

Reply:
מה הקשר?

Reply:
תקשיב אני לא יודע מה לא הבנת... אם באמת אתה רוצה להבין אז תדבג את זה. ככה מונגו עובד שעושים join, הוא מחזיר מערך

Reply:
אני לא מבין אותך
אתה מדבר result[0]?
כי אני לא, תסתכל על איזה שורה אני מדבר
אני אומר שלמערך עובדים אתה צריך לדחוף רק את המנהל וזהו. בלי קונקט